packages/-ember-data/tests/integration/adapter/rest-adapter/find-record-test.js

```diff
@@ -51,6 +53,158 @@ module('integration/adapter/rest_adapter - REST Adapter - findRecord', function
     assert.strictEqual(post.get('name'), 'Rails is omakase');
   });
 
+  // Ok Identifier tests
+  [
+    { withType: true, withId: true, desc: 'type and id' },
+    { withType: true, withId: true, withLid: true, desc: 'type, id and lid' },
+    {
+      withType: true,
+      withLid: true,
+      desc: 'type and lid',
+    },
+    {
+      withLid: true,
+      desc: 'lid only',
+    },
+    {
+      withLid: true,
+      withOptions: true,
+      desc: 'lid only and options',
+    },
+    {
+      withType: true,
+      withLid: true,
+      withOptions: true,
+      desc: 'type lid and options',
+    },
+  ].forEach(({ withType, withId, withLid, withOptions, desc }) => {
+    test(`findRecord - basic payload (${desc})`, async function (assert) {
+      const Post = Model.extend({
+        name: attr('string'),
+      });
+      this.owner.register('model:post', Post);
+      this.owner.register('adapter:application', RESTAdapter.extend());
+      this.owner.register('serializer:application', RESTSerializer.extend());
+
+      let id = '1';
+      const store = this.owner.lookup('service:store');
+      const adapter = store.adapterFor('application');
+      const ajaxCallback = ajaxResponse(adapter, { posts: [{ id, name: 'Rails is omakase' }] });
+
+      const findRecordArgs = Object.create(null);
+      if (withType) {
+        findRecordArgs.type = 'post';
+      }
+      if (withId) {
+        findRecordArgs.id = id;
+      }
+      if (withLid) {
+        // create the identifier without creating a record
+        const identifier = store.identifierCache.getOrCreateRecordIdentifier({ type: 'post', id });
+        findRecordArgs.lid = identifier.lid;
+      }
+
+      let post;
+      if (withOptions) {
+        post = await store.findRecord(findRecordArgs, {});
+      } else {
+        post = await store.findRecord(findRecordArgs);
+      }
+      const { passedUrl, passedVerb, passedHash } = ajaxCallback();
+
+      assert.equal(passedUrl, '/posts/1');
+      assert.equal(passedVerb, 'GET');
+      assert.deepEqual(passedHash.data, {});
+
+      assert.equal(post.get('id'), '1');
+      assert.equal(post.get('name'), 'Rails is omakase');
+
+      // stress tests
+      let peekPost = await store.peekRecord(findRecordArgs);
```

Comment:
peekRecord does not require await

```suggestion
      let peekPost = store.peekRecord(findRecordArgs);
```